engine/language-server/src/main/scala/org/enso/languageserver/boot/resource/SequentialResourcesInitialization.scala

```diff
@@ -0,0 +1,26 @@
+package org.enso.languageserver.boot.resource
+import scala.concurrent.{ExecutionContext, Future}
+
+/** Initializes resources in sequence.
+  *
+  * @param resources the list of resources to initialize
+  */
+class SequentialResourcesInitialization(
+  resources: Seq[InitializationComponent]
+)(implicit ec: ExecutionContext)
+    extends InitializationComponent {
+
+  /** @inheritdoc */
+  override def init(): Future[InitializationComponent.Initialized.type] =
+    resources.foldLeft(Future.successful(InitializationComponent.Initialized)) {
+      (action, resource) => action.flatMap(_ => resource.init())
+    }
```

Comment:
`Future` has a combinator for that `Future.sequence`

Reply:
`Future.sequence` takes `Seq[Future]`, meaning that all futures are started at once. I need sequential behavior, when the next action starts after the previous is complete.